lint-scan.php

```diff
@@ -256,145 +243,99 @@ function vipgoci_lint_scan_commit(
 		$options['skip-draft-prs']
 	);
 
```

Comment:
Thanks for the suggestion. That looks okay. I'm checking its implementation.

Reply:
After a few debug iterations, I could see that:

>if ( true === $options['lint-scan-only-modified-files'] ) {
  $files_to_scan = .... // Get paths to modified files in all the pull requests implicated.
}

- To get the paths to modified files in all the PRs implicated, for each PR, the bot would need to iterate over the $prs_implicated and execute a GET request to ``/repos/:owner/:repo/pulls/:pull_number/files`` or compare the refs that are available in the $prs_implicated objects (sequence of git commands found [here](https://github.com/Automattic/vip-go-ci/pull/212/files/4b51fbd65732a2313fa10c6a46790234c4c9e5fb#diff-8a7186458f366d2d94ecd363f6f23db1b629ffba464ff2821fa3961f7cd57b9fR574));

----

> // Construct array of modified files by PR.
if ( true === $options['lint-scan-only-modified-files'] ) {
   foreach ( $prs_implicated as $pr_item ) {
     $files_changed_in_pr[ $pr_item->number ] = ... // Get files changed in this PR
   }
}

- This would perform better if it was outside of the files iteration (``foreach ( $files_to_scan as $file_name ) {...``);

In the end, if we are to address the comments above, the resulted structure, after breaking functions, would be pretty similar to the code implemented in the original PR. 

From my perspective, the rules are different for both scenarios, so it would make sense to use different functions for each case and reuse functions that are equal for both cases, like scanning files.
  
When we try to solve two different (similar but different) problems with the same code, we tend to add more if conditions, which can affect the application's performance and efficiency.

Reply:
>  From my perspective, the rules are different for both scenarios, so it would make sense to use different functions for each case and reuse functions that are equal for both cases, like scanning files.

It is totally fine to use a different strategy. :slightly_smiling_face:  I only wanted to suggest a different solution.